File: tool/icon_tool/bin/pad_icon.py
#!/usr/bin/env python3
"""Generate a padded adaptive foreground icon from app_icon.png"""

from PIL import Image
import os
import sys

def main():
    src_path = 'assets/icon/app_icon.png'
    out_path = 'assets/icon/app_icon_fg.png'
    
    if not os.path.exists(src_path):
        print(f"Source icon not found: {src_path}")
        sys.exit(2)
    
    # Load image
    img = Image.open(src_path).convert('RGBA')
    
    # Canvas size and content ratio
    canvas_size = 1024
    content_ratio = 0.60
    
    # Calculate target size preserving aspect ratio
    max_content = int(canvas_size * content_ratio)
    w, h = img.size
    
    if w > max_content or h > max_content:
        scale = max_content / max(w, h)
        w = int(w * scale)
        h = int(h * scale)
        img = img.resize((w, h), Image.Resampling.LANCZOS)
    
    # Create transparent canvas
    canvas = Image.new('RGBA', (canvas_size, canvas_size), (0, 0, 0, 0))
    
    # Center and paste
    dx = (canvas_size - img.width) // 2
    dy = (canvas_size - img.height) // 2
    
    canvas.paste(img, (dx, dy), img)
    
    # Ensure output dir exists
    os.makedirs(os.path.dirname(out_path) or '.', exist_ok=True)
    
    canvas.save(out_path)
    print(f"Wrote padded foreground icon to {out_path}")

if __name__ == '__main__':
    main()

File: tool/icon_tool/bin/pad_icon.dart
import 'dart:io';
import 'package:image/image.dart';

void main(List<String> args) {
  final srcPath = 'assets/icon/app_icon.png';
  final outPath = 'assets/icon/app_icon_fg.png';

  if (!File(srcPath).existsSync()) {
    print('Source icon not found: $srcPath');
    exit(2);
  }

  final bytes = File(srcPath).readAsBytesSync();
  Image? src = decodeImage(bytes);
  if (src == null) {
    print('Failed to decode image');
    exit(3);
  }

  const int canvasSize = 1024;
  const double contentRatio = 0.60;

  // Korrektur: In v4 heißt es numChannels, aber der Konstruktor braucht oft explizite Formate
  final Image canvas = Image(
    width: canvasSize, 
    height: canvasSize,
    numChannels: 4,
  );
  
  // Korrektur: ColorRgba8 ist korrekt, aber wir nutzen direkt das Color-Objekt
  fill(canvas, color: ColorRgba8(0, 0, 0, 0));

  final int maxContent = (canvasSize * contentRatio).round();
  
  // Korrektur: Interpolation wird jetzt über das Enum gesteuert
  final Image resized = copyResize(
    src, 
    width: maxContent, 
    height: maxContent, 
    interpolation: Interpolation.cubic,
    maintainAspect: true 
  );

  // Zentrieren
  final int dx = ((canvasSize - resized.width) ~/ 2);
  final int dy = ((canvasSize - resized.height) ~/ 2);

  // Korrektur: compositeImage Parameter heißen in v4 dstX und dstY
  compositeImage(canvas, resized, dstX: dx, dstY: dy);

  // Output speichern
  final outFile = File(outPath);
  outFile.createSync(recursive: true);
  outFile.writeAsBytesSync(encodePng(canvas));

  print('Erfolgreich: Foreground Icon erstellt unter $outPath');
}